### PR TITLE
Smart reply improvements

### PR DESCRIPTION
**Pull request by MrSquirrelDeDuck**
Improves smart reply to fix ping replies being off and it not replying, also made it reraise any exceptions that are not from it being unable to reply.

### CODE REVIEW

File: bread/utility.py [hunk 1]
```diff
@@ -162,17 +162,23 @@ def sanitize_ping(string: str) -> str:
...
-    else:
-        try: 
-            message = await ctx.send(message, mention_author=False)
-        except:
+        else:
+            # Ping reply is off, so send the message and put the author's display name at the top, so they know it's theirs.
             message = await ctx.send(f"{sanitize_ping(ctx.author.display_name)}:\n\n{message}")
+    except:
+        # Some other exception has occurred, so reraise it.
+        raise
```
Comment (fonji) [inline bread/utility.py:182]: isn't this the default behavior? 
I don't know python that well, but in other programming languages, any uncatched exception would bubble up automagically

Comment (fonji) [reply bread/utility.py:182]:
```suggestion
```
:bricks:

**Comment by MrSquirrelDeDuck** [comment]
Yes, it would raise any other exception normally. However in my opinion having it reraise makes it easier to read and understand, since it is very clear what will happen to other exceptions.

**Comment by fonji** [comment]
Interesting, you do you!
Isn't it raising a generic exception instead of a specific one though?

**Pull request merged**